components/butterfly/src/rumor/dat_file.rs

```diff
@@ -527,6 +527,7 @@ mod tests {
     /// This has to actually touch the file system because the nature of the bug its testing
     /// for is Windows-specific: AtomicWriter will fail its rename if the file is held open
     /// by the existence of a BufReader<File>.
+    #[allow(clippy::unnecessary_unwrap)]
```

Comment:
OPTIONAL: rather than add this `allow` which will apply to the entire test, I'd suggest changing the offending line:
```diff
-        assert!(result.is_ok(), "{}", result.unwrap_err());
+        assert!(result.is_ok(), "{:?}", result);
```

Reply:
Ok, will do